llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1483,14 +1483,15 @@ static bool simplifyBranchConditionForVFAndUF(VPlan &Plan, ElementCount BestVF,
     // Try to simplify the branch condition if TC <= VF * UF when the latch
     // terminator is   BranchOnCount or BranchOnCond where the input is
     // Not(ActiveLaneMask).
-    const SCEV *TripCount =
-        vputils::getSCEVExprForVPValue(Plan.getTripCount(), SE);
-    assert(!isa<SCEVCouldNotCompute>(TripCount) &&
+    const SCEV *VectorTripCount =
+        vputils::getSCEVExprForVPValue(&Plan.getVectorTripCount(), SE);
+    if (isa<SCEVCouldNotCompute>(VectorTripCount))
+      VectorTripCount = vputils::getSCEVExprForVPValue(Plan.getTripCount(), SE);
+    assert(!isa<SCEVCouldNotCompute>(VectorTripCount) &&
```

Comment:
Just to check, when do we get a vector trip count that isn't computable but the scalar trip count is? Is it when we have a scalable VF?

Reply:
> We don't have a vector trip count when there is a scalar tail.

Should that be the other way round, i.e we don't have a vector trip count when there is no scalar tail? The tests in this PR that are affected seem to have scalar tails

Reply:
Yes, we get a lot of assertions with the TC not being found.